src/renderers/shared/stack/event/EventPropagators.js

```diff
@@ -11,22 +11,22 @@
 
 'use strict';
 
-var EventConstants = require('EventConstants');
 var EventPluginHub = require('EventPluginHub');
 var EventPluginUtils = require('EventPluginUtils');
 
 var accumulateInto = require('accumulateInto');
 var forEachAccumulated = require('forEachAccumulated');
 var warning = require('warning');
 
-var PropagationPhases = EventConstants.PropagationPhases;
+import type { PropagationPhases } from 'EventConstants';
+
 var getListener = EventPluginHub.getListener;
 
 /**
  * Some event types have a notion of different registration names for different
  * "phases" of propagation. This finds listeners by a given phase.
  */
-function listenerAtPhase(inst, event, propagationPhase) {
+function listenerAtPhase(inst, event, propagationPhase: PropagationPhases) {
   var registrationName =
     event.dispatchConfig.phasedRegistrationNames[propagationPhase];
```

Comment:
@vjeux 

To be specific, the problem is on this line.
GCC will crush [lines like these](https://github.com/vjeux/react/blob/02afcecd643603ec47d882424c48c9ff5663e067/src/renderers/dom/client/eventPlugins/SimpleEventPlugin.js#L38-L43):

```js
{
  abort: {
    phasedRegistrationNames: {
      bubbled: keyOf({onAbort: true}),
      captured: keyOf({onAbortCapture: true}),
    },
  },
}
```

into

```js
{
  a: {
    b: {
      c: keyOf({e: true}),
      d: keyOf({f: true}),
    },
  },
}
```

However, `phase` will be a full string:

```js
var phase = upwards ? 'bubbled' : 'captured';
```

and so

```js
 var registrationName =
      event.dispatchConfig.phasedRegistrationNames[propagationPhase];		     
```

will give you `undefined`.